cdap-app-fabric/src/main/java/co/cask/cdap/app/preview/PreviewStatus.java

```diff
@@ -30,13 +32,15 @@
     RUNNING,
     COMPLETED,
     DEPLOY_FAILED,
-    RUN_FAILED
+    RUN_FAILED,
+    KILLED
   }
 
   private final Status status;
-  private final Throwable throwable;
+  // BasicThrowable is required here so that the exception can be serialized in REST response.
+  private final BasicThrowable throwable;
```

Comment:
Serialization shouldn't change the data type. You can use a Gson codec if necessary. BTW, `BasicThrowable` is not a very good name after all (e.g. is there a AdvancedThrowable??)